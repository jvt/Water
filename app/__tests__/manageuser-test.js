import {
	Text,
	TextInput
} from 'react-native';
import React from 'react';
import Button from 'apsl-react-native-button';
import SegmentedControlTab from 'react-native-segmented-control-tab';
import Renderer from 'react-test-renderer';

describe('ManageUser', () => {

	test('TitleText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			My Account
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('UsernameText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Username:
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('UsernameInput', () => {
		const instance  = Renderer.create(
			<TextInput
			style={{}}
			Username
			/>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('HomeAddressText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Home Address:
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('HomeAddressInput', () => {
		const instance  = Renderer.create(
			<TextInput
			style={{}}
			HomeAddress
			/>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('RoleText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Role:
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('UserTypePicker', () => {
		const instance  = Renderer.create(
			<SegmentedControlTab 
			style={{}}
			Role
			/>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('TitleText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Title:
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('TitleInput', () => {
		const instance  = Renderer.create(
			<TextInput
			style={{}}
			Title
			/>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('SaveButton', () => {
		const instance  = Renderer.create(
			<Button 
			style={{}}>
			Save
			</Button>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('ReturnButton', () => {
		const instance  = Renderer.create(
			<Button 
			style={{}}>
			Return
			</Button>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

});
