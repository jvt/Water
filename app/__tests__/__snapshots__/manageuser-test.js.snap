exports[`ManageUser HomeAddressInput 1`] = `
<TextInput
  HomeAddress={true}
  style={Object {}} />
`;

exports[`ManageUser HomeAddressText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Home Address:
</Text>
`;

exports[`ManageUser ReturnButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Return
  </Text>
</View>
`;

exports[`ManageUser RoleText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Role:
</Text>
`;

exports[`ManageUser SaveButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Save
  </Text>
</View>
`;

exports[`ManageUser TitleInput 1`] = `
<TextInput
  Title={true}
  style={Object {}} />
`;

exports[`ManageUser TitleText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  My Account
</Text>
`;

exports[`ManageUser TitleText 2`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Title:
</Text>
`;

exports[`ManageUser UserTypePicker 1`] = `
<View
  removeClippedSubviews={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "row",
      },
      Object {},
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0076FF",
        "borderBottomLeftRadius": 5,
        "borderColor": "#0076FF",
        "borderTopLeftRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {
            "color": "white",
          },
          undefined,
        ]
      }>
      One
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#0076FF",
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {},
          Object {},
        ]
      }>
      Two
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomRightRadius": 5,
        "borderColor": "#0076FF",
        "borderTopRightRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
          undefined,
          Object {},
          Object {},
        ]
      }>
      Three
    </Text>
  </View>
</View>
`;

exports[`ManageUser UsernameInput 1`] = `
<TextInput
  Username={true}
  style={Object {}} />
`;

exports[`ManageUser UsernameText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Username:
</Text>
`;
